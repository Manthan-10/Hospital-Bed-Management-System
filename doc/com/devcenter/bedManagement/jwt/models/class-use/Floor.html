<!DOCTYPE HTML>
<html lang="en">
<head>
<!-- Generated by javadoc (18) on Mon May 30 12:53:08 IST 2022 -->
<title>Uses of Class com.devcenter.bedManagement.jwt.models.Floor</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2022-05-30">
<meta name="description" content="use: package: com.devcenter.bedManagement.jwt.models, class: Floor">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon"></span><span class="nav-bar-toggle-icon"></span><span class="nav-bar-toggle-icon"></span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Class com.devcenter.bedManagement.jwt.models.Floor" class="title">Uses of Class<br>com.devcenter.bedManagement.jwt.models.Floor</h1>
</div>
<div class="caption"><span>Packages that use <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#com.devcenter.bedManagement.jwt.controllers">com.devcenter.bedManagement.jwt.controllers</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#com.devcenter.bedManagement.jwt.models">com.devcenter.bedManagement.jwt.models</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#com.devcenter.bedManagement.jwt.repository">com.devcenter.bedManagement.jwt.repository</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#com.devcenter.bedManagement.jwt.security.services">com.devcenter.bedManagement.jwt.security.services</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.controllers">
<h2>Uses of <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a> in <a href="../../controllers/package-summary.html">com.devcenter.bedManagement.jwt.controllers</a></h2>
<div class="caption"><span>Methods in <a href="../../controllers/package-summary.html">com.devcenter.bedManagement.jwt.controllers</a> with parameters of type <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FloorController.</span><code><a href="../../controllers/FloorController.html#createFloor(com.devcenter.bedManagement.jwt.models.Floor)" class="member-name-link">createFloor</a><wbr>(<a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a>&nbsp;floor)</code></div>
<div class="col-last even-row-color">
<div class="block">Function createFloor is mapped to /api/test/floor/addfloor for adding floors</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.models">
<h2>Uses of <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a> in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a></h2>
<div class="caption"><span>Methods in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a> that return <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Room.</span><code><a href="../Room.html#getFloor()" class="member-name-link">getFloor</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a> with parameters of type <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Room.</span><code><a href="../Room.html#setFloor(com.devcenter.bedManagement.jwt.models.Floor)" class="member-name-link">setFloor</a><wbr>(<a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a>&nbsp;floor)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a> with parameters of type <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../Room.html#%3Cinit%3E(int,java.lang.String,com.devcenter.bedManagement.jwt.models.Floor)" class="member-name-link">Room</a><wbr>(int&nbsp;id,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;roomNo,
 <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a>&nbsp;floor)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.repository">
<h2>Uses of <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a> in <a href="../../repository/package-summary.html">com.devcenter.bedManagement.jwt.repository</a></h2>
<div class="caption"><span>Methods in <a href="../../repository/package-summary.html">com.devcenter.bedManagement.jwt.repository</a> that return types with arguments of type <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FloorRepository.</span><code><a href="../../repository/FloorRepository.html#getFloorsByHospital(com.devcenter.bedManagement.jwt.models.Hospital)" class="member-name-link">getFloorsByHospital</a><wbr>(<a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a>&nbsp;hosp)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../repository/package-summary.html">com.devcenter.bedManagement.jwt.repository</a> with parameters of type <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RoomRepository.</span><code><a href="../../repository/RoomRepository.html#getRoomsByFloor(com.devcenter.bedManagement.jwt.models.Floor)" class="member-name-link">getRoomsByFloor</a><wbr>(<a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a>&nbsp;floor)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.security.services">
<h2>Uses of <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a> in <a href="../../security/services/package-summary.html">com.devcenter.bedManagement.jwt.security.services</a></h2>
<div class="caption"><span>Methods in <a href="../../security/services/package-summary.html">com.devcenter.bedManagement.jwt.security.services</a> that return types with arguments of type <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FloorService.</span><code><a href="../../security/services/FloorService.html#getFloors()" class="member-name-link">getFloors</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Function for fetching all floors</div>
</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FloorService.</span><code><a href="../../security/services/FloorService.html#getFloorsByHospital(java.lang.Integer)" class="member-name-link">getFloorsByHospital</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;id)</code></div>
<div class="col-last odd-row-color">
<div class="block">Function which fetches the details of all the floors for the given Hospital id</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../security/services/package-summary.html">com.devcenter.bedManagement.jwt.security.services</a> with parameters of type <a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FloorService.</span><code><a href="../../security/services/FloorService.html#createFloor(com.devcenter.bedManagement.jwt.models.Floor)" class="member-name-link">createFloor</a><wbr>(<a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a>&nbsp;floor)</code></div>
<div class="col-last even-row-color">
<div class="block">Function for adding new floors</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
</div>
</div>
</body>
</html>
