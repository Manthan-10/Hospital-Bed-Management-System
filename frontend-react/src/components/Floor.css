.rooms{
    margin-top:0;
}