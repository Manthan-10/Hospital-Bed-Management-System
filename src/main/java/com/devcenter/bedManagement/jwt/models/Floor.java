package com.devcenter.bedManagement.jwt.models;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;

@Entity
@Table(name = "Floor")
public class Floor {
	

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	
	private String floorNo;
	
	@ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name="hospital_id", referencedColumnName="id")
    private Hospital hospital;
	
	
	@OneToMany(mappedBy="floor", cascade = CascadeType.ALL)
	private List<Room> rooms;
	
	
	public Floor() {
		super();
	}


	public Floor(int id, String floorNo, Hospital hospital) {
		super();
		this.id = id;
		this.floorNo = floorNo;
		this.hospital = hospital;
	}


	public int getId() {
		return id;
	}


	public void setId(int id) {
		this.id = id;
	}
	
	



	

	public String getFloorNo() {
		return floorNo;
	}


	public void setFloorNo(String floorNo) {
		this.floorNo = floorNo;
	}


	public Hospital getHospital() {
		return hospital;
	}


	public void setHospital(Hospital hospital) {
		this.hospital = hospital;
	}

	
	
	
	

}
