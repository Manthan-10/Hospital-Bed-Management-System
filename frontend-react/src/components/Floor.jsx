import React from "react";
import { useParams } from "react-router";
import Hospital from "./Hospital";
import Room from "./Room";
import "./Floor.css";

/**
 * This component displays the floor in a particular Hospital
 *
 */
const Floor = () => {
  const { fid } = useParams();

  return (
    <div>
      <Hospital />
      <div className="rooms">
        <Room fid={fid} />
      </div>
    </div>
  );
};

export default Floor;
