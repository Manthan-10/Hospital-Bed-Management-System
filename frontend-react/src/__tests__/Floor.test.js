import React from 'react';
import Enzyme,{ shallow, mount } from 'enzyme';
import Floor from '../components/Floor';
import Adapter from 'enzyme-adapter-react-16';
import * as reactRouterDom from "react-router-dom";
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() })

/**
 * Testing for UI of Login page
 * @author MN097789
 */
 

 const mockNavigate = jest.fn();

 jest.mock("react-router-dom", () => ({
 
   useSelector: jest.fn(),
 
   useNavigate: () => mockNavigate,
 
 }));
 const mockNavigateSpy = jest.spyOn(reactRouterDom, "useNavigate");
 const initialState = { output: 10 };
    const mockStore = configureStore();
    let store;

 
 describe('Floor UI Test Suite', () => {

    let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Floor />);
  });

  test('Checks if it renders correctly', () => {
    store = mockStore(initialState);
    const tree = renderer.create(<Provider store={store}><Floor /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('should check for divs', () => {
    wrapper = shallow(<Floor />);
    expect(wrapper.length).toEqual(1);
  

    });
});