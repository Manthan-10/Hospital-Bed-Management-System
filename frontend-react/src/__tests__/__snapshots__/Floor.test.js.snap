// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Floor UI Test Suite Checks if it renders correctly 1`] = `
<div>
  <div
    className="hospital"
  >
    <div
      className="hospital-wrapper"
    >
      <div
        className="hospital-dropdown"
      >
        <div
          className="hsp-dropdown"
        >
          <select
            className="select-container"
            onChange={[Function]}
            size="large"
          >
            <option>
              Select_Hospital
            </option>
          </select>
        </div>
      </div>
      <div
        className="floor-dropdown"
      >
        <div
          className="floors"
        >
          <select
            className="select-container"
            onChange={[Function]}
            size="large"
          >
            <option>
              Select_Floor
               
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="rooms"
  >
    <div
      className="room"
    >
      <div
        className="headings"
      />
      <div
        className="room-wrapper"
      />
    </div>
  </div>
</div>
`;
